### PR TITLE
Fix version information in docker images

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area quality
/kind bug
/priority normal

**What this PR does / why we need it**:
Before, we were always building our docker images with `version.gitTreeState=dirty`.
This PR ignores changes to the following files while checking the git tree state during build time:
- `VERSION`: the version file is currently the only way to inject the version-to-build in our pipelines (see https://github.com/gardener/cc-utils/issues/431). Hopefully, this can be replaced by setting the `EFFECTIVE_VERSION` docker build arg in our pipelines in the future.
- all files ignored by `.dockerignore`: these files are not present in the build container -> git detects them as deleted

The PR also removes the `v`-Prefix from `version.major`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement developer
The version information in docker images has been updated to correctly display `version.major` without the `v`-Prefix and `version.gitTreeState` as `clean`.
```

### CODE REVIEW

**Comment by timebertt** [comment]
/hold
Let me checkout the images built in this pipeline run to do some additional verification before merging :)

**Comment by timebertt** [comment]
The images built by the pipeline looks good wrt to the `gitTreeState`.
But I noticed another issue while checking: the `major` attribute has the `v` prefix:
```
$ docker run -p 8443:8443 ... eu.gcr.io/gardener-project/gardener/apiserver:v1.8.0-dev-82226631efda5a8d1fba2d4e8e7c75ddd186f5fb ... &
...
$ curl -k https://localhost:8443/version
{
  "major": "v1",
  "minor": "8",
  "gitVersion": "v1.8.0-dev-82226631efda5a8d1fba2d4e8e7c75ddd186f5fb",
  "gitCommit": "82226631efda5a8d1fba2d4e8e7c75ddd186f5fb",
  "gitTreeState": "clean",
  "buildDate": "2020-07-21T06:26:15+00:00",
  "goVersion": "go1.14.5",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

Let me fix that behaviour and add the fix to this PR.
/status author-action

**Comment by timebertt** [comment]
Fixed the prefix issue.
Now the version information is set correctly as expected:

```
$ docker run -p 8443:8443 ... eu.gcr.io/gardener-project/gardener/apiserver:v1.8.0-dev-53a6d1e75d0c57b61e35951ebff5529733a56b56 ... &
...
$ curl -k https://localhost:8443/version
{
  "major": "1",
  "minor": "8",
  "gitVersion": "v1.8.0-dev-53a6d1e75d0c57b61e35951ebff5529733a56b56",
  "gitCommit": "53a6d1e75d0c57b61e35951ebff5529733a56b56",
  "gitTreeState": "clean",
  "buildDate": "2020-07-21T10:21:33+00:00",
  "goVersion": "go1.14.5",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

/unassign
/remove status/author-action
/unhold
/needs review

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**
